Frameworks/UIKit/StarboardXaml/StarboardXaml.cpp

```diff
@@ -95,14 +95,28 @@ void App::OnActivated(IActivatedEventArgs^ args) {
             _ApplicationMainLaunch(ActivationTypeToast, argsString);
         }
         UIApplicationMainHandleToastNotificationEvent(Strings::WideToNarrow(argsString->Data()).c_str());
+    } else if (args->Kind == ActivationKind::VoiceCommand) {
+        Windows::Media::SpeechRecognition::SpeechRecognitionResult^ argResult = safe_cast<VoiceCommandActivatedEventArgs^>(args)->Result;
+        TraceVerbose(TAG, L"Received voice command with argument - %s", argResult->Text->Data());
+        if (initiateAppLaunch) {
+            _ApplicationMainLaunch(ActivationTypeVoiceCommand, argResult);
+        }
+        UIApplicationMainHandleVoiceCommandEvent(reinterpret_cast<IInspectable*>(argResult));
+    } else if (args->Kind == ActivationKind::Protocol) {
+        Windows::Foundation::Uri^ argUri = safe_cast<ProtocolActivatedEventArgs^>(args)->Uri;
+        TraceVerbose(TAG, L"Received protocol with uri- %s", argUri->ToString()->Data());
+        if (initiateAppLaunch) {
+            _ApplicationMainLaunch(ActivationTypeProtocol, argUri);
+        }
+        UIApplicationMainHandleProtocolEvent(reinterpret_cast<IInspectable*>(argUri));
     } else {
```

Comment:
I think I forgot to a add a trace here, can we add one so we know when the application gets activated it was due to an unhandled activation kind?